### PR TITLE
FSET-1731: Schemes without evaluation should be progressed to the right state

### PR DESCRIPTION
**Pull request by fayimora**
Some schemes do not require an evaluation(no pass/fail). If these schemes
are the green schemes after sift or form submission, the candidate should be
progressed to the right state.

### CODE REVIEW

**Comment by henricook** [comment]
Conflicts @fayimora

**Comment by fayimora** [comment]
@henricook Fixed!

File: app/services/sift/SiftAnswersService.scala [hunk 1]
```diff
@@ -75,10 +75,25 @@ trait SiftAnswersService {
...
   }
+
+  private def maybeMoveToCompleted(applicationId: String, passedSchemes: Set[SchemeId],
+                                   passedSchemesWithoutRequirement: Set[SchemeId]): Future[Unit] = {
+    val canBeMoved = (passedSchemes.size == passedSchemesWithoutRequirement.size) &&
+      passedSchemes.forall(passedSchemesWithoutRequirement.contains)
+    if(canBeMoved) {
+      appRepo.addProgressStatusAndUpdateAppStatus(applicationId, ProgressStatuses.SIFT_COMPLETED)
+    } else {
+      Future(())
```
Comment (benjstephenson) [inline app/services/sift/SiftAnswersService.scala]: return future success explicitly or you'll create another thread.

File: app/services/sift/SiftAnswersService.scala [hunk 2]
```diff
@@ -75,10 +75,25 @@ trait SiftAnswersService {
...
       ).map(_.id).toSet
       _ <- siftAnswersRepo.submitAnswers(applicationId, schemesPassedRequiringSift)
       _ <- appRepo.addProgressStatusAndUpdateAppStatus(applicationId, ProgressStatuses.SIFT_READY)
+      _ <- maybeMoveToCompleted(applicationId, schemesPassed, schemesPassedWithoutRequirement)
     } yield {}
   }
+
+  private def maybeMoveToCompleted(applicationId: String, passedSchemes: Set[SchemeId],
+                                   passedSchemesWithoutRequirement: Set[SchemeId]): Future[Unit] = {
+    val canBeMoved = (passedSchemes.size == passedSchemesWithoutRequirement.size) &&
```
Comment (benjstephenson) [inline app/services/sift/SiftAnswersService.scala]: belt and braces, but - we should really be checking that the sets are the same, not just the same size.

File: app/services/sift/SiftAnswersService.scala [hunk 3]
```diff
@@ -75,10 +75,25 @@ trait SiftAnswersService {
...
+      passedSchemes.forall(passedSchemesWithoutRequirement.contains)
```
Comment (benjstephenson) [inline app/services/sift/SiftAnswersService.scala]: Use set equality rather than iterating over them.

**Comment by benjstephenson** [comment]
@fayimora a few comments - Can you also write some unit tests please.

**Pull request merged**